cmd/provenanced/cmd/cmd_test.go

```diff
@@ -24,3 +25,60 @@ func TestInitCmd(t *testing.T) {
 	err := cmd.Execute(rootCmd)
 	require.NoError(t, err)
 }
+
+func TestGenAutoCompleteCmd(t *testing.T) {
+	home := t.TempDir()
+
+	tests := []struct {
+		name string
+		args []string
+		err  string
+	}{
+		{
+			name: "failure - missing arg",
+			err:  "accepts 1 arg(s), received 0",
+		},
+		{
+			name: "failure - too many args",
+			args: []string{"bash", "fish"},
+			err:  "accepts 1 arg(s), received 2",
+		},
+		{
+			name: "failure - invalid shell type",
+			args: []string{"badshellname"},
+			err:  "shell badshellname is not supported",
+		},
+		{
+			name: "success - works with bash",
+			args: []string{"bash"},
+		},
+		{
+			name: "success - works with zsh",
+			args: []string{"zsh"},
+		},
+		{
+			name: "success - works with fish",
+			args: []string{"fish"},
+		},
+		{
+			name: "success - works with powershell",
+			args: []string{"powershell"},
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			args := []string{"--home", home, "enable-cli-autocomplete"}
+			args = append(args, tc.args...)
+
+			rootCmd, _ := cmd.NewRootCmd(false)
+			rootCmd.SetArgs(args)
+			err := cmd.Execute(rootCmd)
+			if len(tc.err) > 0 {
+				assertions.AssertErrorValue(t, err, tc.err, "should have correct error")
+			} else {
+				require.NoError(t, err, "should not output error on command execution")
+			}
```

Comment:
```suggestion
			assertions.AssertErrorValue(t, err, tc.err, "enable-cli-autocomplete error")
```

The `if len(tc.err) > 0` and `else` part is built into `assertions.AssertErrorValue`.

The `msgAndArgs` part should describe how the variable being checked was made; in this case it's the `err` variable that is created by running the `enable-cli-autocomplete` command. That way, when looking at the failure message, you know more about what triggered the error (or lack of error) that you're looking at in the message.

Whether an error was or was not expected is redundant in the `msgAndArgs`. When any of `EqualError`, `NoError`, or `AssertErrorValue` fails, the failure message tells you whether either a) an error was expected and not received, b) an error was received that was different from expected, or c) an error was not expected, but was received.

Many tests have several `NoError` requirements that happen during setup, so it's really helpful for the `msgAndArgs` to indicate what was done to create the error (even if the test only has one assertion). It removes ambiguity, so it's more likely that you can identify what's going wrong by just looking at the test failure message. Usually, it's good enough to include the function name, but if that same function is being used multiple times, it's also good to include extra information to distinguish one call/check from the other. For commands, though, it's good to use the command name instead of the function being run (`Execute` or whatever) since that function isn't very descriptive.

I know the failure messages will include `TestGenAutoCompleteCmd` in the output, but when looking at a test failure message, you can't assume that `GenAutoCompleteCmd` is what was called to create the thing causing the failure. That's another reason why it's good to provide a `msgAndArgs` that describes the thing being checked.